luda-editor/luda-editor-client/src/app/sequence_list/types/sequence_index.rs

```diff
@@ -0,0 +1,122 @@
+        &self,
+        title_sequence_map: &LinkedHashMap<String, T>,
+    ) -> LinkedHashMap<String, T>
+    where
+        T: Clone,
+    {
+        let mut sorted_title_sequence_map: LinkedHashMap<String, T> = LinkedHashMap::new();
+        for title in &self.sequences {
+            title_sequence_map.get(title).and_then(|sequence| {
+                sorted_title_sequence_map.insert(title.clone(), sequence.clone())
+            });
+        }
+        for (title, sequence) in title_sequence_map {
+            if sorted_title_sequence_map.contains_key(title) {
+                continue;
+            }
+            sorted_title_sequence_map.insert(title.clone(), sequence.clone());
+        }
+        sorted_title_sequence_map
+    }
+
+    pub async fn save(&self, socket: &Socket) -> Result<(), String> {
+        socket
+            .write_file(luda_editor_rpc::write_file::Request {
+                dest_path: SEQUENCE_INDEX_PATH.to_string(),
+                file: serde_json::to_vec(&self.sequences).unwrap(),
+            })
+            .await
+            .and_then(|_| Ok(()))
+    }
+
+    pub async fn load(socket: &Socket) -> Self {
+        match socket
+            .read_file(luda_editor_rpc::read_file::Request {
+                dest_path: SEQUENCE_INDEX_PATH.to_string(),
+            })
+            .await
+            .and_then(|response| {
+                let sequences: Result<Vec<String>, serde_json::Error> =
+                    serde_json::from_slice(&response.file);
+                match sequences {
+                    Ok(sequences) => Ok(sequences),
+                    Err(error) => Err(error.to_string()),
+                }
+            }) {
+            Ok(sequences) => Self { sequences },
+            Err(error) => {
+                namui::log!("{}", error);
+                Self { sequences: vec![] }
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::app::sequence_list::types::SequenceIndex;
+    use linked_hash_map::LinkedHashMap;
+    use wasm_bindgen_test::wasm_bindgen_test;
+
+    #[test]
+    #[wasm_bindgen_test]
+    fn sort_indexed_first_then_concat_rest() {
+        let index: Vec<String> = vec!["4".to_string(), "2".to_string()];
+        let mut map: LinkedHashMap<String, bool> = LinkedHashMap::new();
+        map.insert("1".to_string(), true);
+        map.insert("2".to_string(), true);
+        map.insert("3".to_string(), true);
+        map.insert("4".to_string(), true);
+
+        let sorted_map = SequenceIndex::new(index).sort_title_sequence_map(&map);
+        let sorted_keys: Vec<String> = sorted_map.keys().map(|key| key.clone()).collect();
+
+        // input:  [1, 2, 3, 4]
+        // index:  [4, 2]
+        // 1. sort indexed first   [4, 2]  [1, 3]
+        // 2. then concat rest     [4, 2, 1, 3]
+        // expected:   [4, 2, 1, 3]
+        let expected_keys: Vec<String> = vec![
+            "4".to_string(),
+            "2".to_string(),
+            "1".to_string(),
```

Comment:
I think 1 and 3 must not in map because it's not in index

Reply:
Sequence that does not exist in the spreadsheet and that exists locally as file will not appear.
Is this what you intend?

Reply:
Yes and actually that local file should be deleted. Spreadsheet is source of truth, and local is kind of mirror. 

There are some trouble that we don't know it just renamed or removed. 

In this PR, I hope we only care about spreadsheet's list.
And then, at next PR, please check that google spreadsheet's sheet has id or not, then use id instead of title for json filename and put sheet name at top of json file.